src/store/songs/actions.js

```diff
@@ -0,0 +1,16 @@
+import { CHANGE_VOTE_FAIL } from './constants'
+
+export const changeVote = song => (dispatch, getState, getFirebase) =>
```

Comment:
So this basically is called with a specific song, and returns a function with how to changeVotes for that specific song? Is there a reason we dont dispatch anything on the success case?

Reply:
yes, we're using a redux middleware that is automatically dispatching actions and managing updates to the `firebase` slice of redux state in response to firebase data changing. if we were managing that part of state ourselves, we would define our own actions to describe those state updates and dispatch them.